checkstyle/checks.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" ?>
+<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<module name="Checker">
+        <property name="charset" value="UTF-8" />
+
+        <property name="severity" value="error" />
+
+        <property name="fileExtensions" value="java, properties, xml" />
+
+        <!-- Checks for whitespace                               -->
+        <!-- See http://checkstyle.org/config_whitespace.html -->
+        <module name="FileTabCharacter">
+                <property name="eachLine" value="true" />
+        </module>
+
+        <!-- https://checkstyle.org/config_filters.html#SuppressionFilter -->
+        <module name="SuppressionFilter">
+                <property name="file" value="checkstyle/suppressions.xml" />
+        </module>
```

Comment:
Do we want to keep suppressions in another file? I think it would be a lot better to implement this using [comment suppression](https://checkstyle.sourceforge.io/config_filters.html#SuppressionCommentFilter), which is far less prone to going out of date.

```suggestion
        <module name="TreeWalker">
        	<!-- https://checkstyle.org/config_filters.html#SuppressionCommentFilter -->
        	<module name="SuppressionCommentFilter" />
        </module>
```